src/data-handler.ts

```diff
@@ -0,0 +1,39 @@
+import axios from 'axios';
+import { config } from "./config";
+import { Course } from './Model/Course';
+import { CourseParams } from './util/types';
+
+export async function fetchAndStoreCourses() {
+    try {
+      const authResponse = await axios.post('https://metergram-courses-api.azurewebsites.net/api/auth', {
+        apiKey: config.api_key,
+      });
+
+    console.log(authResponse.data);
+      
+      const accessToken = authResponse.data.data.accessToken;
```

Comment:
Make separate file/functions for requests to external API. Handle the errors there, read the URL from config, handle the response data, (the first .data comes from axios), so that you don't have .data.data etc... expose functions for fetching courses, for auth etc...